sys/trickle/trickle.c

```diff
@@ -18,9 +18,11 @@
 #include "random.h"
 #include "trickle.h"
 
-#define ENABLE_DEBUG    (0)
+#define ENABLE_DEBUG        (0)
 #include "debug.h"
 
+#define MIN(a,b)            ((a < b) ? a : b)
```

Comment:
using a macro with a calculation is bad, since the preprocessor will produce
```c
trickle->I = (trickle->I < (trickle->Imin << trickle->Imax)) ? trickle->Imin : (trickle->Imin << trickle->Imax)) 
```
So it's completely compiler dependent to optimize both calculations into a single one and reuse the register.
I am not a fan of macros of this sort  and the current implementation stores the result explicitely in `max_interval`. What is wrong with using `max_interval`? It's not that memory (on the stack) is wasted..